core/arch/riscv/riscv.mk

```diff
@@ -71,9 +71,20 @@ core-platform-cppflags	+= -I$(arch-dir)/include
 core-platform-subdirs += \
 	$(addprefix $(arch-dir)/, kernel mm tee) $(platform-dir)
 
-# more convenient to move it to platform instead
-rv64-platform-cppflags += -mcmodel=medany -march=rv64imafd -mabi=lp64d
+# Default values for "-mcmodel", "-march", and "-abi" compiler flags.
+# Platform-specific overrides are in core/arch/riscv/plat-*/conf.mk.
+riscv-platform-mcmodel ?= medany
+rv64-platform-isa ?= rv64imafd
+rv64-platform-abi ?= lp64d
+rv32-platform-isa ?= rv32imafd
+rv32-platform-abi ?= ilp32d
+
+rv64-platform-cppflags += -mcmodel=$(riscv-platform-mcmodel)
+rv64-platform-cppflags += -march=$(rv64-platform-isa) -mabi=$(rv64-platform-abi)
 rv64-platform-cppflags += -Wno-missing-include-dirs
+rv32-platform-cppflags += -mcmodel=$(riscv-platform-mcmodel)
+rv32-platform-cppflags += -march=$(rv32-platform-isa) -mabi=$(rv32-platform-abi)
+rv32-platform-cppflags += -Wno-missing-include-dirs
```

Comment:
Why is this needed?

Reply:
@jenswi-linaro Resolved in third commit.